marlowe-playground-client/src/MonadApp.purs

```diff
@@ -229,33 +229,32 @@ withMarloweEditor = HalogenApp <<< Editor.withEditor _marloweEditorSlot unit
 
 updateContractInStateP :: String -> MarloweState -> MarloweState
 updateContractInStateP text state = case runParser' (parseTerm contract) text of
-  Right pcon ->
+  Right parsedContract ->
     let
-      lintResult = lint pcon
-
-      lintHoles = view L._holes lintResult
-
-      (Holes holes) = lintHoles
+      lintResult = lint parsedContract
 
       warnings =
         map (warningToAnnotation text "The contract can make a negative payment here") (view L._negativePayments lintResult)
           <> map (warningToAnnotation text "The contract can make a negative deposit here") (view L._negativeDeposits lintResult)
+          <> map (warningToAnnotation text "Timeouts should always increase in value") (view L._timeoutNotIncreasing lintResult)
+          <> map (warningToAnnotation text "The contract tries to Use a ValueId that has not been defined in a Let") (view L._uninitializedUse lintResult)
+          <> map (warningToAnnotation text "Let is redfining a ValueId that already exists") (view L._shadowedLet lintResult)
+          <> map (warningToAnnotation text "This Observation will always evaluate to True") (view L._trueObservation lintResult)
+          <> map (warningToAnnotation text "This Observation will always evaluate to False") (view L._falseObservation lintResult)
 
```

Comment:
What do you think about making this a call like this?

``` purescript
foldMap foo warningTypes
where
  warningTypes = [ ("The contract can make ...", L._negativePayments)
                 , ("Timeouts should always...", L._timeoutNotIncreasing)
                 , ...
                 ]
```

Reply:
I tried it out and it doesn't really look that good